src/addrman.h

```diff
@@ -715,14 +723,20 @@ friend class CAddrManTest;
         return addrRet;
     }
 
-    //! Return a bunch of addresses, selected at random.
-    std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct)
+    /**
+     * Return all or many randomly selected addresses, optionally by network.
+     *
+     * @param[in] max_addresses  Maximum number of addresses to return (0 = all).
+     * @param[in] max_pct        Maximum percentage of addresses to return (0 = all).
+     * @param[in] network        Select only addresses of this network (nullopt = all).
+     */
+    std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct, std::optional<Network> network = std::nullopt)
```

Comment:
It just occurred to me that the only (non test) caller of `GetAddr()` is `GetAddresses()` which will always pass 3 arguments. Maybe consider removing the default from here since it is not necessary.

```suggestion
    std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct, std::optional<Network> network)
```

Reply:
Good idea, doing (this will add fuzz coverage for change too).